build: update angular and bazel

gkalpak opened the pull request:
This is a combination of #44779 and #45224 (with some additional fixes to make CI pass).

mary-poppins commented:
You can preview 0f44357 at https://pr45235-0f44357.ngbuilds.io/.

mary-poppins commented:
You can preview 559acf1 at https://pr45235-559acf1.ngbuilds.io/.

mary-poppins commented:
You can preview 703072d at https://pr45235-703072d.ngbuilds.io/.

mary-poppins commented:
You can preview 27faa0b at https://pr45235-27faa0b.ngbuilds.io/.

mary-poppins commented:
You can preview e4dc7b8 at https://pr45235-e4dc7b8.ngbuilds.io/.

mary-poppins commented:
You can preview be63991 at https://pr45235-be63991.ngbuilds.io/.

renovate-bot committed:
build: update bazel

Update `@bazel` packages to the latest 5.x version.

Some of the changes here are modeled after
angular/dev-infra@40c0ac855975ce9586b07af3a36a0bf4e3b58350.

renovate-bot committed:
build: update angular to 5a82e60

Update the `@angular/dev-infra-private` package to the latest commit.

mary-poppins commented:
You can preview 4c87e75 at https://pr45235-4c87e75.ngbuilds.io/.

mary-poppins commented:
You can preview b772e80 at https://pr45235-b772e80.ngbuilds.io/.

mary-poppins commented:
You can preview ed94fb1 at https://pr45235-ed94fb1.ngbuilds.io/.

devversion commented on WORKSPACE:
```diff
@@ -34,33 +38,41 @@ http_archive(
+    name = "nodejs",
     node_version = "16.10.0",
-    package_json = ["//:package.json"],
 )
```
You may also want to set up `yarn_install` explicitly like in dev-infra (to use the vendored-one)

mary-poppins commented:
You can preview d378df7 at https://pr45235-d378df7.ngbuilds.io/.

mary-poppins commented:
You can preview 4de2ab8 at https://pr45235-4de2ab8.ngbuilds.io/.

devversion committed:
refactor(bazel): update ngc-wrapped to account for `tsc-wrapped` move to `@bazel/concatjs`

Previously `tsc-wrapped` which is the foundation for `ngc-wrapped`, resided
in `@bazel/typescript`. It has been moved to `@bazel/concatjs` in rules_nodejs
so we need to account for that as part of our rules_nodejs v5 update.

mary-poppins commented:
You can preview 512581c at https://pr45235-512581c.ngbuilds.io/.

devversion committed:
fixup! build: update bazel

fixes a mismatch of the `ts_project` rule. The one for `ts_project` was accidentally
used in `ts_library` causing build failures.

devversion committed:
build: add temporary patch to make `tsec` compatible with `rules_nodejs` v5

Adds a temporary patch to make `tsec` compatible with `rules_nodejs` v5
until https://github.com/google/tsec/pull/25 is available/released.

mary-poppins commented:
You can preview 4447727 at https://pr45235-4447727.ngbuilds.io/.

devversion committed:
refactor(bazel): update api-extractor to account for `@bazel/typescript` change

`@bazel/typescript` code moved to `@bazel/concatjs` for the tsc-wrapped code.

Note that this code is likely going to be removed anyway soon when we
move dts bundling from `ng_module` to `ng_package`.

mary-poppins commented:
You can preview 6d32a54 at https://pr45235-6d32a54.ngbuilds.io/.

devversion committed:
build: migrate more usages from `@bazel/typescript` to `@bazel/concatjs`

As mentioned in previous commits (check them for more details), `@bazel/typescript`
no longer contains `ts_library`-specific code, so we no longer need that dependency.

mary-poppins commented:
You can preview 113a323 at https://pr45235-113a323.ngbuilds.io/.

mary-poppins commented:
You can preview a12e7ee at https://pr45235-a12e7ee.ngbuilds.io/.

mary-poppins commented:
You can preview 44f9560 at https://pr45235-44f9560.ngbuilds.io/.

devversion committed:
refactor: remove unused variables in starlark code to satisfy buildifier

We updated buildifier and a few warnings became errors now. This commit
cleans up the failing unused variable instances, making the linter happy.

Additionally for the API extractor BUILD file, the package defaults
need to move to satisfy buildifier.

devversion committed:
ci: temporarily disable components-repo-unit-tests job for Bazel NodeJS v5 update

The Angular components repository can only start using Bazel Rules NodeJS v5
when `@angular/bazel` is published with support for it. To work around this
cycle we temporarily disable the unit tests job until we migrated the COMP
repo as well.

mary-poppins commented:
You can preview 332fb23 at https://pr45235-332fb23.ngbuilds.io/.

mary-poppins commented:
You can preview 5e25a77 at https://pr45235-5e25a77.ngbuilds.io/.

devversion committed:
ci: make windows circleci job more robust and use git bash

It is totally fine, and expected to use Git Bash for running Bazel
on Windows. In fact this is the most common setup for Bazel on Windows
and it's unrealistic to run without it.

This allows us to remove the old/legacy Powershell setup from CI
which is also quite flaky sometimes and does not reproduce how
Bazel is used on windows-users dev machines.

mary-poppins commented:
You can preview ccb1d14 at https://pr45235-ccb1d14.ngbuilds.io/.

mary-poppins commented:
You can preview 1668747 at https://pr45235-1668747.ngbuilds.io/.

mary-poppins commented:
You can preview e41e83b at https://pr45235-e41e83b.ngbuilds.io/.

mary-poppins commented:
You can preview 79269eb at https://pr45235-79269eb.ngbuilds.io/.

mary-poppins commented:
You can preview 48b0ad3 at https://pr45235-48b0ad3.ngbuilds.io/.

mary-poppins commented:
You can preview 0201952 at https://pr45235-0201952.ngbuilds.io/.

josephperrott reviewed (approved):
LGTM

mary-poppins commented:
You can preview 6862c2d at https://pr45235-6862c2d.ngbuilds.io/.

mary-poppins commented:
You can preview 6e87b77 at https://pr45235-6e87b77.ngbuilds.io/.

mary-poppins commented:
You can preview 8b50dfd at https://pr45235-8b50dfd.ngbuilds.io/.

devversion committed:
test: refactor compiler-cli compliance test to work on windows

Recent changes in `rules_nodejs` caused the test case copy file actions
to be transitioned into the `exec` configuration, resulting in much larger
file paths. These paths break on Windows with the shell argument limit, and
with the path limit, causing errors like:

```
ERROR: C:/users/circleci/ng/packages/compiler-cli/test/compliance/test_cases/BUILD.bazel:9:12: Copying file packages/compiler-cli/test/compliance/test_cases/r3_compiler_compliance/components_and_directives/value_composition/structural_directives_if_directive_def.js failed: (Exit 1): cmd.exe failed: error executing command
  cd /d C:/users/circleci/_bazel_circleci/u4uoan2j/execroot/angular
  SET PATH=C:\Program Files\Git\usr\bin;C:\Program Files\Git\bin;C:\Windows;C:\Windows\System32;C:\Windows\System32\WindowsPowerShell\v1.0
    SET RUNFILES_MANIFEST_ONLY=1
  cmd.exe /C bazel-out\x64_windows-opt-exec-2B5CBBC6\bin\packages\compiler-cli\test\compliance\test_cases\test_cases--1973427149-cmd.bat
The system cannot find the path specified
```

https://app.circleci.com/pipelines/github/angular/angular/44038/workflows/4b530cb2-f232-4e1d-b35a-e6e085151d08/jobs/1140017

gkalpak commented:
Superseded by #45431.
